pkgs/development/compilers/ghc/head.nix

```diff
@@ -12,7 +12,7 @@ let
 
   commonBuildInputs = [ ghc perl autoconf automake happy alex python3 ];
 
-  version = "8.1.20170106";
+  version = "2017-01-06";
```

Comment:
The ghc HEAD version numbers as used by the upstream (generated by git, e.g. ghc-8.3-start-421-g43a31683ac) are not used at all here.

To explain once again, my change is based on
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23"

from https://nixos.org/nixpkgs/manual/#sec-package-naming.